src/extension/background-script/connectors/alby.ts

```diff
@@ -311,10 +317,9 @@ export default class Alby implements Connector {
 
       throw error;
     } finally {
-      const newToken = this._authUser.token;
-      if (newToken && newToken !== oldToken) {
-        await this._updateOAuthToken(newToken);
-      }
+      this._authUser.on("tokenRefreshed", async (token: Token) => {
```

Comment:
shouldn't we define this event listener once globally whenever we set the `this._authUser`?  (as I think you do above)
and then remove this finally.

Reply:
my bad! I missed the link, this will also duplicate the event listener.

so the flow would be.
1. user creates the account, event is registered as it's registered in authorize function.
2. if token is expired when making a request, token will be updated, and callback is called registered via event listener.